### PR TITLE
python3 improvements

### PR DESCRIPTION
**Pull request by mastizada**
Part of https://wiki.mozilla.org/L10n:Pontoon/Python_3 work.

### CODE REVIEW

**Comment by mastizada** [comment]
@jsocol test fails but code looks working correctly. In `eq_('2.3.4.5', req.META['REMOTE_ADDR'])` as I read the code, add 127.0.0.1 to the REMOTE_ADDR and then adds that ip to the end of the list `ips`, `ips.reverse()` makes it first ip and then in the loop, as 127.0.0.1 not in known ips it returns that.

**Comment by jsocol** [comment]
Hey @mastizada, thanks for the patch!

I'm not maintaining this library anymore—I'm actually surprised anyone even within Mozilla is still using it. I'm happy to transfer it into the Mozilla org if it's still in use, since they're the copyright holders, and add some owners to PyPI. I reached out to the #webdev channel in Mozilla IRC, but if you can help route that to the right folks, it'd be helpful.

FWIW—and this is from very old memories—I think I'd put `127.0.0.1` into `KNOWN_PROXIES` _if there was a reverse proxy running on localhost_. If you're running nginx or haproxy locally in front of Django, and it's connecting from localhost, then there should be a localhost entry in the list. So I believe this behavior is correct but something about the test, settings, or test client is weird. That said, I'm not 100% sure why this only just started failing, but I'm guessing it's [new behavior in Django](https://github.com/jsocol/commonware/pull/22/files#diff-b4ef698db8ca845e5845c4618278f29aL5) since 1.6.

The test matrix almost certainly needs to be updated: Django doesn't support Py2.6 anymore, and does support a bunch of new versions of Py3. You might want to look at dropping 2.6 first to clear that build failure up. It's probably also useful to run the tests against multiple Django versions, like the ESR and the current supported releases. That might help nail down where the issue was introduced.

**Comment by mastizada** [comment]
Yes, that makes sense, will add tox to test different django versions and update known proxies.

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ pip-log.txt
 *.egg-info
 dist
 .coverage
+venv/
+src/
```
Comment (mastizada) [inline .gitignore]: used by pip install for plugins from git

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 language: python
 python:
- - "2.6"
  - "2.7"
-install: pip install -r requirements.txt --use-mirrors
```
Comment (mastizada) [inline .travis.yml]: use-mirrors is deprecated in python2.7

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
-script: fab test
+ - "3.6"
+install: pip install tox-travis
```
Comment (mastizada) [inline .travis.yml]: use tox in local, tox-travis matches python version in travis

File: commonware/response/middleware.py [hunk 4]
```diff
@@ -43,14 +43,14 @@ class StrictTransportMiddleware(object):
     """
     Set the Strict-Transport-Security header on responses. Use the
     STS_MAX_AGE setting to control the max-age value. (Default: 1 year.)
-    Use the STS_SUBDOMAINS boolean to add includeSubdomains.
+    Use the SECURE_HSTS_INCLUDE_SUBDOMAINS boolean to add includeSubdomains.
```
Comment (mastizada) [inline commonware/response/middleware.py]: Couldn't find any reference for `STS_SUBDOMAINS`, Django docs suggests using `SECURE_HSTS_INCLUDE_SUBDOMAINS` for `includeSubdomains`

File: fabfile.py [hunk 5]
```diff
@@ -14,8 +14,7 @@
 ROOT = os.path.abspath(os.path.dirname(__file__))
 
 os.environ['DJANGO_SETTINGS_MODULE'] = '%s-project.settings' % NAME
-os.environ['PYTHONPATH'] = os.pathsep.join([ROOT,
-                                            os.path.join(ROOT, 'examples')])
+os.environ['PYTHONPATH'] = os.pathsep.join([ROOT, os.path.join(ROOT, 'examples')])
```
Comment (mastizada) [inline fabfile.py]: django style guide suggests 119 line length

File: requirements.txt [hunk 6]
```diff
@@ -1,7 +1,6 @@
 # These are required to run the tests.
 nose
 mock
-fabric
-Django<1.7
```
Comment (mastizada) [inline requirements.txt]: Django will be added by tox, to prevent duplicate requirement it was removed from txt file.

**Comment by mastizada** [comment]
For some reason, `settings._wrapped` doesn't change settings (that error with known proxies setting). Using `settings` instead solved problem for all python versions.

File: commonware/response/middleware.py (hunk 4 above)
Comment (jsocol) [inline commonware/response/middleware.py]: pretty sure this is all redundant with django built-in features now

File: fabfile.py (hunk 5 above)
Comment (jsocol) [inline fabfile.py]: as not-the-owner-anymore, I don't have a strong opinion, but style guide changes are usually not changes that should be made lightly in the middle of another PR. that needs to change the linter rules, and should be discussed separately

**Comment by jsocol** [comment]
Hey there @mastizada—moved the repo into the Mozilla org and gave the team owner writes on PyPI. They'll need to review the code at the end of the day—hopefully I've been helpful in getting it moved along. Thanks again!

File: commonware/response/middleware.py (hunk 4 above)
Comment (pmac) [inline commonware/response/middleware.py]: Yup. This should be removed and the docs updated to point to [Django's `SecurityMiddleware`](https://docs.djangoproject.com/en/1.8/ref/middleware/#security-middleware).

File: contribute.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "Commonware",
+    "description": "A collection of small but useful tools for Django.",
+    "repository": {
+        "url": "https://github.com/jsocol/commonware",
```
Comment (pmac) [inline contribute.json]: Has moved to `https://github.com/mozilla/commonware`. Also 👍 for adding this file!! 🥇

File: commonware/response/middleware.py (hunk 4 above)
Comment (mastizada) [inline commonware/response/middleware.py]: Added Deprecated exception with url to SecurityMiddleware doc.

**Comment by pmac** [comment]
I've squashed and merged this. I should have a release ready soon. Thanks!

**Comment by pmac** [comment]
Just an FYI: I've just released `v0.5.0` which includes these changes.

**Pull request closed without merging**